devicetypes/sky-nie/in-wall-smart-switch-dimmer.src/in-wall-smart-switch-dimmer.groovy

```diff
@@ -0,0 +1,692 @@
+/**
```

Comment:
Since all those 4 DTH share some capabilities, please create one DTH that will support all Your devices.
You may use different vid for Your device to hide or expose capability supported by certain device.
You may need to add some if statements and tweaks in the code in Your methods to support behavior or every device, but that way we will avoid the code duplication, because there are many exact the same methods in every DTH You submitted here.

Reply:
For writing DTH files, I am a beginer. The examples I have seen so far include "tiles" and "attribute". If it is not supported in the future, where can I see new examples or help documents?

Reply:
It's important to add **vid and mnmn fields in the fingerprints** (depending on what capability Your devices support), it can be for example one of the following

**for switches:**
- mnmn: "SmartThings", vid: "vid:"generic-switch" // if Your device doesn't report [W] and [kWh]
- mnmn: "SmartThings", vid: "vid:"generic-switch-energy"  // if Your device doesn't report [W] and reports [kWh]
- mnmn: "SmartThings", vid: "vid:"generic-switch-power"  // if Your device reports [W] and doesn't report [kWh]
- mnmn: "SmartThings", vid: "vid:"generic-switch-power-energy" // if Your device reports [W] and [kWh]

**for switches/dimmers:**
- mnmn: "SmartThings", vid: "vid:"generic-dimmer" // if Your device doesn't report [W] and [kWh]
- mnmn: "SmartThings", vid: "vid:"generic-dimmer-power"  // if Your device reports [W] and doesn't report [kWh]
- mnmn: "SmartThings", vid: "vid:"generic-dimmer-power-energy" // if Your device reports [W] and [kWh]

but keep in mind that those energy reports have to be handled in the DTH.
You may also need to add one of these:
- ocfDeviceType: "oic.d.switch"
- ocfDeviceType: "oic.d.smartplug" 

to the fingerprint (depending on if Your device is a smart plug or a switch or switch/dimmer)

Reply:
Thank you, I am studying how to merge these files and how to use vid to simplify the code.